lib/manifestManager/index.js

```
@@ -0,0 +1,28 @@
+export class ManifestManager {
+  static $inject = ['MANIFEST_PATH'];
+  static $singleton = true;
+
+  constructor(manifestPath) {
+    if(!manifestPath) {
+      throw 'Can\'t init ManifestManger instance without passing manifestPath as a param.';
+    }
+    this.manifestPath = manifestPath;
+    this.setManifest();
+  }
+
+  setManifest() {
+      if(!this._manifest) {
+        this._manifest = require(this.manifestPath);
+      }
+
+      console.log(`Manifest successfully has been set. ${this._manifest}`);
+  }
+
+  get(manifestProp) {
+    if(!this._manifest) {
+      return console.error(`Trying to get ${manifestProp} prop of not existed manifest. Try to call setManifest method on an instance.`);
+    }
+
+    return this._manifest[manifestProp] || console.error(`There's no ${manifestProp} prop found. ${Object.keys(this._manifest)}`);
```

I'm a bit confused with this expression: `return this._manifest[manifestProp] || console.error(....)`

---

In case there's no needed prop in manifest we will return undefined and console.error the act of attempt to get not existed prop. Does it seems relevant behavior?

---

Let's just throw exception in case of undefined